apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -35,7 +41,18 @@ pub(super) fn validate_selection(
     schema: &SchemaInfo,
     seen_fields: &mut IndexSet<(Name, Name)>,
 ) -> Result<(), Message> {
-    let (selection_arg, json_selection) = get_json_selection(coordinate, schema)?;
+    let (selection_arg, json_selection, selection_node) = get_json_selection(coordinate, schema)?;
+
+    if let Some(value) = validate_selection_variables(
```

Comment:
just curious: why this pattern and not `if let Err(error) = validate_selection_variables(...) { return Err(error) }`?

Reply:
Ugh, I refactored this code out to a method using the IDE refactoring, and it did... whatever this is. I missed that weird `Option<Result<...>>`. Will fix.